debian/clickhouse-server.service

```diff
@@ -10,7 +10,7 @@ Group=clickhouse
 Restart=always
 RestartSec=30
 RuntimeDirectory=clickhouse-server
-ExecStart=/usr/bin/clickhouse-server --config=/etc/clickhouse-server/config.xml --pid-file=/run/clickhouse-server/clickhouse-server.pid
+ExecStart=/usr/bin/clickhouse-server --config=/etc/clickhouse-server/config.xml --pid-file=/var/run/clickhouse-server/clickhouse-server.pid
```

Comment:
Have you check this? I'm pretty sure that there another issue and this will not solve it (in fact looks like it can make things worse)

Reply:
Which command do you have in cron?

Reply:
I guess that the problem is that `clickhouse-server.init` (i.e. script for sysv-init) does not know that there is systemd used.
And this is because `clickhouse-server.init` from upstream does not detect is systemd used or not, while debian version does:
- https://salsa.debian.org/debian/ClickHouse/-/blob/dacfb19f68d51f27a900318abc2cbd99f1b412ea/debian/clickhouse-server.init#L39
- https://salsa.debian.org/systemd-team/systemd/-/blob/fa278866b50f15486583eba39d5c4427c0ef20dd/debian/extra/init-functions.d/40-systemd#L46

So I think that [clickhouse-server.init](https://github.com/ClickHouse/ClickHouse/blob/master/debian/clickhouse-server.init) should be patched to detect systemd, and if so, proxy to it like debian sysvinit scripts does.

Reply:
@azat You are right. I misunderstood the issue.  I will close the PR and open another one once I fix the issue. Thanks for your comments.

Reply:
And I wouldn't recommend using cron with systemd, take a look at:
- [systemd.timer](https://www.freedesktop.org/software/systemd/man/systemd.timer.html)
- [Restart](https://www.freedesktop.org/software/systemd/man/systemd.service.html#Restart=)

Reply:
>@azat You are right. I misunderstood the issue. I will close the PR and open another one once I fix the issue.

@amitsharma10 are you still planning to do this?

Reply:
Right now [clickhouse-server.init](https://github.com/ClickHouse/ClickHouse/blob/master/debian/clickhouse-server.init) (that is in clickhouse repository) has extra complexity and includes some issues:
- does not uses start-stop-daemon (and has pretty complex functions with manual kill/pgrep/reading pid file)
- does not detect is system uses systemd (while debian based systemd unit have this detection, and this is kind of standard there)

And there are some issues reported with this.

@alexey-milovidov What about just replacing clickhouse-server.init with the unit from the [debian repository](https://salsa.debian.org/debian/ClickHouse/-/blob/dacfb19f68d51f27a900318abc2cbd99f1b412ea/debian/clickhouse-server.init#L39)? Or maybe there are some reasons for this complexity?

Reply:
I don't like all the complexity, especially if it's located inside shell scripts and if it cannot be tested.
We can replace our init script if the new script will provide comparable usability, specifically:
- forcestop and forcerestart commands that are doing kill -9;
- messages about successfult start/stop.

Reply:
Our script has support for systemd, see: https://github.com/ClickHouse/ClickHouse/blob/master/debian/clickhouse-server.init#L258

Reply:
> What about just replacing clickhouse-server.init with the unit from the debian repository?

No, we cannot do that because our init script should work on non-Debian OS.

Reply:
>Our script has support for systemd, see: https://github.com/ClickHouse/ClickHouse/blob/master/debian/clickhouse-server.init#L258

But there is no support for start/stop/... only for cond*/forcerestart, am I missing something?

Reply:
>cron enable/disable.

Looks very non standard

>No, we cannot do that because our init script should work on non-Debian OS.

I though about this LSB thing, ok, then it cannot be simply replaced

Reply:
> But there is no support for start/stop/... only for cond*/forcerestart, am I missing something?

Don't know why, let's fix it.

Reply:
https://github.com/ClickHouse/ClickHouse/pull/14460

Reply:
> Our script has support for systemd, see: https://github.com/ClickHouse/ClickHouse/blob/master/debian/clickhouse-server.init#L258

It should call systemctl, not service

Reply:
Ok (I have no idea what is the difference - you now better) - let's change the script.

Reply:
>It should call systemctl, not service

In practice it should make any different (except for bypassing extra script), since `service` will call `systemctl` [directly (at least in debian)](https://salsa.debian.org/debian/init-system-helpers/-/blob/6a1860c6f5ad295af605ddf588933544e7c24ce1/script/service#L187)

Reply:
> I have no idea what is the difference - you know better

* systemctl is a native binary, core part of systemd. 
* service is a wrapper shell script which runs either init.d either systemd service (has some logic to pick which one to use). 

> since service will call systemctl directly (at least in debian) 

1) Why one more hop of redirections since we checked that systemctl executable exists? 
2) here is how service script look on centos: https://gist.github.com/filimonov/998fa2b46496669c5c965cfed818dc34